client/me/developer/main.jsx

```diff
@@ -26,8 +28,62 @@ class Developer extends Component {
 		recordTracksEvent( 'calypso_me_is_dev_account_toggled', {
 			enabled: isDevAccount ? 1 : 0,
 		} );
+
+		const cookies = cookie.parse( document.cookie );
+
+		if ( isDevAccount && ! [ 'completed', 'dismissed' ].includes( cookies.developer_survey ) ) {
+			const noticeId = 'developer-survey';
+
+			this.props.successNotice(
+				this.props.translate(
+					"Hey developer! How do {{i}}you{{/i}} use WordPress.com? Spare a moment? We'd love to hear what you think in a {{surveyLink}}quick survey{{/surveyLink}}.",
+					{
+						components: {
+							i: <i />,
+							surveyLink: (
+								<a
+									href="handle_me"
+									target="_blank"
+									rel="noopener noreferrer"
+									onClick={ () => {
+										recordTracksEvent( 'calypso_me_developer_survey_clicked' );
+
+										this.props.removeNotice( noticeId );
+									} }
+								/>
+							),
+						},
+					}
+				),
+				{
+					id: noticeId,
+					isPersistent: true,
+					onDismissClick: () => {
+						recordTracksEvent( 'calypso_me_developer_survey_dismissed' );
+
+						document.cookie = cookie.serialize( 'developer_survey', 'dismissed', {
+							path: '/',
+							maxAge: 24 * 60 * 60, // 1 day
+						} );
```

Comment:
```suggestion
						document.cookie = cookie.serialize( 'developer_survey', 'dismissed', {
							path: '/',
							maxAge: 24 * 60 * 60, // 1 day
						} );
```

I would suggest creating a function for saving the cookie so that we'd only have to define the cookie name and path in one spot.